### PR TITLE
support ntc in dnn layer

### PR DESCRIPTION
**Pull request by zhangxiaoli73**
## What changes were proposed in this pull request?

support ntc in dnn layer

## How was this patch tested?

unit tests

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/XXX

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/DnnGraph.scala [hunk 1]
```diff
@@ -55,7 +55,7 @@ class DnnGraph(
       val out = output.toTensor[Float]
       // for grey image, input should be 3 dims and the first dim should be batch size
       // for non grey image, input should be 4 dims and the first dim should be batch size
-      require(in.nDimension() == 4 || in.nDimension() == 3,
+      require(in.nDimension() == 4 || in.nDimension() == 3 || in.nDimension() == 2,
```
Comment (i8run) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/DnnGraph.scala]: Please add comment about the condition of `in.nDimension() == 2`

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/intermediate/IRGraph.scala [hunk 2]
```diff
@@ -141,7 +141,10 @@ private[bigdl] class IRGraph[T: ClassTag](
       if (input.isInstanceOf[Tensor[T]]) {
         // todo: handle for 3 dimensions, expand 3 dims to 4 dims
         val size = input.toTensor[T].size()
-        val sizeNew = if (size.length == 3)  Array(size(0), 1, size(1), size(2)) else size
+        val sizeNew = if (size.length == 3 && inputFormats(0) != Memory.Format.ntc
+          && inputFormats(0) != Memory.Format.tnc) {
+          Array(size(0), 1, size(1), size(2))
+        } else size
```
Comment (i8run) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/intermediate/IRGraph.scala]: Why we should create `sizeNew` ? Could it be handled in `Input` layer/op ?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/DnnGraph.scala (hunk 1 above)
Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/DnnGraph.scala]: ok, changed

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/intermediate/IRGraph.scala (hunk 2 above)
Comment (zhangxiaoli73) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/intermediate/IRGraph.scala]: input layer needs input shape and format. Dim 3 has no corresponding format, and its actual channel is 1, so here we add one dim and create sizeNew

Comment (i8run) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/intermediate/IRGraph.scala]: OK. Got it.

**Comment by zhangxiaoli73** [comment]
jenkins pass: http://mlt-bddl01.sh.intel.com:8080/view/PR/job/BigDL-PRVN-lite/676/

**Pull request merged**